Form - Vue and React samples (#628)
DevExpress/devextreme-documentation: concepts/05 Widgets/Form/20 Change Options at Runtime/15 Editor Options.md

[note]The **getEditor(dataField)** method is available for visible form items only.

> With Angular, Vue or React, bind the option to change in the [editorOptions](/api-reference/10%20UI%20Widgets/dxForm/5%20Item%20Types/SimpleItem/editorOptions.md '/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/#editorOptions') object to a component or element property.

Reviewer: ```suggestion
With Angular, Vue, or React, bind the option that should be changed in the [editorOptions](/api-reference/10%20UI%20Widgets/dxForm/5%20Item%20Types/SimpleItem/editorOptions.md '/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/#editorOptions') object to a component property.
```

        // ...
    })

##### Vue

    <!-- tab: App.vue -->
    <template>
        <div>
            <DxForm :form-data="employee">
                <DxSimpleItem data-field="firstName" :editor-options="editorOptions" />
                <DxSimpleItem data-field="lastName" />
                <DxSimpleItem data-field="phone" />
                <DxSimpleItem data-field="email" />
            </DxForm>
            <DxCheckBox
                text="Disable the First Name Editor"

>                 :value.sync="isFirstNameDisabled"

Reviewer: ```suggestion
                :value="editorOptions.disabled"
```

                @value-changed="checkBoxValueChanged" />
        </div>
    </template>
    <script>
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

>     import { DxForm, DxCheckBox } from 'devextreme-vue';

Reviewer: There should be no imports directly from 'devextreme-vue' or 'devextreme-react'. Please check that across all these documents.

Author: Fixed

    import { DxSimpleItem } from 'devextreme-vue/form';

    const employee = {
        firstName: 'John',
        lastName: 'Heart',
        phone: '+1(213) 555-9392',
        email: 'jheart@dx-email.com'
    };

    export default {
        components: {
            DxForm, DxSimpleItem, DxCheckBox
        },
        data() {

>             const isFirstNameDisabled = true;

Reviewer: ```suggestion

```

            return {
                employee,

>                 editorOptions : {
>                     disabled: isFirstNameDisabled
>                 }
>                 isFirstNameDisabled

Reviewer: ```suggestion
                editorOptions: {
                    disabled: false
                }
```

            };
        },
        methods: {
            checkBoxValueChanged(e) {
                this.editorOptions = { disabled: e.value };
            }
        }
    };
    </script>

##### React

    <!-- tab: App.js -->
    import React from 'react';

    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import { Form, CheckBox } from 'devextreme-react';
    import { SimpleItem } from 'devextreme-react/form';

    const employee = {
        firstName: 'John',
        lastName: 'Heart',
        phone: '+1(213) 555-9392',
        email: 'jheart@dx-email.com'
    };

    class App extends React.Component {
        constructor() {
            super();

>             const isFirstNameDisabled = true;
>             this.state = {
>                 isFirstNameDisabled,
>                 editorOptions: {
>                     disabled: isFirstNameDisabled
>                 }
>             };

Reviewer: ```suggestion
            this.state = {
                editorOptions: {
                    disabled: false
                }
            };
```

            this.onCheckBoxValueChanged = this.onCheckBoxValueChanged.bind(this);
        };

        render() {
            return (
                <div>
                    <Form formData={employee}>

>                         <SimpleItem dataField="firstName" editorOptions={{disabled: this.state.isFirstNameDisabled}} />

Reviewer: ```suggestion
                        <SimpleItem dataField="firstName" editorOptions={this.state.editorOptions} />
```

                        <SimpleItem dataField="lastName" />
                        <SimpleItem dataField="phone" />
                        <SimpleItem dataField="email" />
                    </Form>
                    <CheckBox
                        text="Show the Phone Number"

>                         value={this.state.isFirstNameDisabled}

Reviewer: ```suggestion
                        value={this.state.editorOptions.disabled}
```

>                         onValueChanged={this.onCheckBoxValueChanged } />

Reviewer: ```suggestion
                        onValueChanged={this.onCheckBoxValueChanged} />
```

                </div>
            );
        }

        onCheckBoxValueChanged(e) {

>             this.setState({
>                 isFirstNameDisabled: e.value,
>                 editorOptions: {
>                     disabled: e.value 
>                 }
>             });

Reviewer: ```suggestion
            this.setState({
                editorOptions: {
                    disabled: e.value 
                }
            });
```

        // ...
    })

##### Vue

    <!-- tab: App.vue -->
    <template>
        <div>
            <DxForm :form-data="employee">

>                 <DxSimpleItem data-field="firstName" :editor-options="{disabled: isFirstNameDisabled}" />

Reviewer: In Vue and React, we do not recommend declaring objects in the markup. I suppose we will have bind editorOption to an object and reassign the entire object.

Author: Fixed https://github.com/DevExpress/devextreme-documentation/pull/628/commits/6fef4f3309f3225bf7c28b6e32ddebf89045804f
